### ISSUE TITLE
Rbspy doesn't work with Ruby 2.1 and 2.2

### ISSUE THREAD

**Issue by Pr0p1k** [rbspy/rbspy#266]
Mac OS X 10.15
I have ruby 2.2.10 and run this command: `sudo rbspy record -- ruby bar.rb`
Then I get this output:
```
Dropping permissions: running Ruby command as user aleksandr.prokopyev
Press Ctrl+C to stop
barabrrbspy(65420,0x70000d363000) malloc: can't allocate region
:*** mach_vm_map(size=1261748260941824, flags: 60000100) failed (error code=3)
rbspy(65420,0x70000d363000) malloc: *** set a breakpoint in malloc_error_break to debug
memory allocation of 1261748260938320 bytes failedAbort trap: 6
```

This is my ruby script: 
```
class Foo
  def bar
    print 'barabr'
  end

  def foo
    10.times do
      bar
      sleep 0.1
    end
  end
end

Foo.new.foo
```

In docker (SYS_PTRACE cap is added) it also fails for ruby 2.1, but the exit code is 134.

**Comment by acj** [comment]
Sorry for the delay here. I was able to repro this crash but haven't found the root cause yet. Some of the thread values we're getting (e.g. `thread.stack_size`) are garbage and look like memory addresses, so my guess is that the thread address is wrong or we're casting to an incorrect type somewhere.

### Repro details

macOS 11.2.3 with ruby 2.2.10

Ruby 2.2.x doesn't build out of the box for me, so a few overrides are needed:

```
brew install rbenv/tap/openssl@1.0
CFLAGS="-Wno-implicit-function-declaration" RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.0) --disable-install-doc" asdf install ruby 2.2.10
```

With extra debug code added:

```
$ sudo RUST_LOG=debug ./target/debug/rbspy record -- ruby ci/ruby-programs/infinite.rb
Dropping permissions: running Ruby command as user acj
Press Ctrl+C to stop
[2021-04-12T21:35:33Z DEBUG rbspy::core::initialize] version: 2.2.10
[2021-04-12T21:35:33Z DEBUG rbspy::core::initialize] current thread address: 4437594440
[2021-04-12T21:35:33Z DEBUG rbspy::core::initialize] VM address: 4437594424
[2021-04-12T21:35:33Z DEBUG rbspy::core::initialize] global symbols address: 4437522800
[2021-04-12T21:35:33Z DEBUG rbspy] Starting main record loop
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] get_stack_trace top
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] if stack_field
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] Stack is at 140285603136240
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] stack_base
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] Stack is at 140285603136240
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] Stack size is 140285602465792
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] get_cfps
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] Stack is at 140285603136240
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] Stack size is 140285602465792
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] stack_base is 1262570422862496, cfp_address is 140285711646720
[2021-04-12T21:35:33Z DEBUG rbspy::core::ruby_version::ruby_2_1_10] vector of size 14028558890197
rbspy(72900,0x70000a7d1000) malloc: can't allocate region
:*** mach_vm_map(size=1122284711219200, flags: 100) failed (error code=3)
rbspy(72900,0x70000a7d1000) malloc: *** set a breakpoint in malloc_error_break to debug
memory allocation of 1122284711215760 bytes failed
```

**Comment by acj** [comment]
Aha, the crash on 2.2.10 is due to a typo in the version string:

https://github.com/rbspy/rbspy/blob/bc8c96995791ea34c2b890023e8fe941a396ce7a/src/core/initialize.rs#L438

I'll fix that for the 0.5 release.

The issue you had on ruby 2.1 may be specific to Docker or the base image you were using. I'm able to run rbspy against ruby 2.1.0 and 2.1.10 without issue on macOS 11 and Linux (Ubuntu 20.10). Exit code 134 is SIGABRT ([ref](https://tldp.org/LDP/abs/html/exitcodes.html)).

I'm going to close this once the typo is fixed, but please feel free to reopen (with more OS/enviroment details) if you're still seeing the issue with 2.1 in Docker.
